server/internal/clients/clients.go

```diff
@@ -217,29 +222,51 @@ func (cl *Client) ForgetSubscription(filter string) {
 
 // Start begins the client goroutines reading and writing packets.
 func (cl *Client) Start() {
-	cl.State.started.Add(2)
+	var started sync.WaitGroup
+
+	started.Add(2)
+	cl.State.endedR.Add(1)
```

Comment:
FYI I think these two Add(1) calls have to move above the corresponding `go` statements. This may contribute to #25, since without Add before the goroutine, it's possible for Stop to be called before one goroutine starts b/c Done can be reached w/o Add.